python/TestHarness/testers/Tester.py

```diff
@@ -41,6 +41,8 @@ def validParams():
     params.addParam('petsc_debug',   ['ALL'], "{False,True} -> test only runs when PETSc is configured with --with-debugging={0,1}, otherwise test always runs.")
     params.addParam('curl',          ['ALL'], "A test that runs only if CURL is detected ('ALL', 'TRUE', 'FALSE')")
     params.addParam('tbb',           ['ALL'], "A test that runs only if TBB is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('pthreads',      ['ALL'], "A test that runs only if PTHREADS is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('openmp',        ['ALL'], "A test that runs only if OPENMP is available ('ALL', 'TRUE', 'FALSE')")
```

Comment:
This test is an example, I am counting calls to initialSetup,
residualSetup, subdomainSetup, etc. in UserObjects.  This problem has two
subdomains and depending on the threading model the way that the elements
and nodes are distributed changes, thus changing the number of calls to
subdomainSetup.

On Tue, Mar 1, 2016 at 8:24 AM, Derek Gaston <notifications@github.com>
wrote:

> In python/TestHarness/testers/Tester.py
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6375-23discussion-5Fr54581802&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=1n52bxzEmzkOwQJ2048t0uTxHgNisOpHwTDiyOPXTmw&s=W1TGMDpqznr6I_O4-v5SA9-wWuFKZ04-ajq1LLLP9Ug&e=>
> :
>
> > @@ -41,6 +41,8 @@ def validParams():
> >      params.addParam('petsc_debug',   ['ALL'], "{False,True} -> test only runs when PETSc is configured with --with-debugging={0,1}, otherwise test always runs.")
> >      params.addParam('curl',          ['ALL'], "A test that runs only if CURL is detected ('ALL', 'TRUE', 'FALSE')")
> >      params.addParam('tbb',           ['ALL'], "A test that runs only if TBB is available ('ALL', 'TRUE', 'FALSE')")
> > +    params.addParam('pthreads',      ['ALL'], "A test that runs only if PTHREADS is available ('ALL', 'TRUE', 'FALSE')")
> > +    params.addParam('openmp',        ['ALL'], "A test that runs only if OPENMP is available ('ALL', 'TRUE', 'FALSE')")
>
> We shouldn't need to check for these in the test harness. The only reason
> to check is if we have a feature that explicitly only works with one model
> (like the NiCE integration stuff relying on TBB). Other than that, we
> ALWAYS have at least one threading model enabled... but it's abstracted out.
>
> Can you guys explain why we would need to ever check for OpenMP or
> Pthreads in the test harness?
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6375_files-23r54581802&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=1n52bxzEmzkOwQJ2048t0uTxHgNisOpHwTDiyOPXTmw&s=b3F3u9gInjwTuh-q_pKa03R8gQ-TwytPI90khTsOoRU&e=>
> .
>